spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/datamining/RowTransformer.scala

```diff
@@ -0,0 +1,324 @@
+  )(implicit ev: TensorNumeric[T]): RowTransformer = {
+    val transSchemas = numericFields.map { case(key, fields) => ColsToNumeric[T](key, fields) }
+    new RowTransformer(transSchemas.toSeq)
+  }
+
+  /**
+   * A `RowTransformer` which contains both `atomic` schemas and `numeric` schemas.
+   *
+   * @param atomicFields field names of `selected columns`
+   * @param numericFields Map<`schemaKey`, `fieldNames of selected columns`> of numeric fields
+   */
+  def atomicWithNumeric[@specialized(Float, Double) T: ClassTag](
+    atomicFields: Seq[String],
+    numericFields: Map[String, Seq[String]]
+  )(implicit ev: TensorNumeric[T]): RowTransformer = {
+    val transSchemas = mutable.ArrayBuffer[RowTransformSchema]()
+    atomicFields.foreach(f => transSchemas += ColToTensor(f, f))
+    numericFields.foreach { case(key, fields) =>
+      transSchemas += ColsToNumeric[T](key, fields)
+    }
+    new RowTransformer(transSchemas)
+  }
+
+}
+
+/**
+ *  A `schema` describe a transforming job which convert a `Row` to a `Table`(`Tensor`).
+ */
+trait RowTransformSchema extends Serializable {
+
+  /**
+   *  Key of the schema, which will be the key of `Tensor` in result `Table`.
+   *  So, it should be `unique` in single `RowTransformer`.
+   */
+  val schemaKey: String
+
+  /**
+   *  `Indices` of Selected Columns
+   *  It will work on only when `fieldNames` is empty,
+   *  otherwise `RowTransformer` will select columns accord to `fieldNames`.
+   *  If both `indices` and `fieldNames` are empty,
+   *  `RowTransformer` will select all columns by default.
+   */
+  val indices: Seq[Int] = Seq.empty
+
+  /**
+   *  `FieldNames` of Selected Columns
+   *  This property will override `indices` when it is not empty.
+   */
+  val fieldNames: Seq[String] = Seq.empty
+
+  /**
+   *  Transforming Logic of the Schema
+   *
+   *  @param values values of selected columns
+   *  @param fields StructFields of selected columns
+   *  @return a result `Tensor`
+   */
+  def transform(values: Seq[Any], fields: Seq[StructField]): Tensor[NumericWildcard]
+
+}
+
+/**
+ * A schema which specialize on transforming multiple `numeric` columns to one `Tensor`.
+ * Types of `selected columns` will be identified according to their `[DataType`.
+ * And type conversions will be done automatically from `DataType` to `T` if valid.
+ * Currently, `DoubleType`, `FloatType`, `ShortType`, `IntegerType`, `LongType` are supported.
+ *
+ * @param schemaKey key of the schema
+ * @param indices indices of `selected columns`
+ * @param fieldNames field names of `selected columns`
+ * @tparam T the type of result `Tensor`
+ */
+class ColsToNumeric[@specialized(Float, Double) T: ClassTag](
+  override val schemaKey: String,
+  override val indices: Seq[Int] = Seq.empty,
+  override val fieldNames: Seq[String] = Seq.empty
+)(implicit ev: TensorNumeric[T]) extends RowTransformSchema {
+
+  override def transform(input: Seq[Any], fields: Seq[StructField]): Tensor[NumericWildcard] = {
```

Comment:
Looks like the return type is Tensor[T]?

Reply:
@yiheng  Because its super method returns a Tensor[NumericWildcard] for universality.

Reply:
I see